### PR TITLE
Feature/co 612 update terms api to dw1.1.0

### PR DESCRIPTION
**Pull request by gmtcrary**
This PR integrates the changes in the web-api-skeleton as of commit 9489e63043369d682a1d16ec5d0a792eed045034

It upgrades the Dropwizard version from 0.8.2 to 1.1.0

Authentication setup in TermsApplication.groovy is the largest change within the terms-api code itself.

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/termsapi/TermsApplication.groovy [hunk 1]
```diff
@@ -47,12 +44,15 @@ class TermsApplication extends Application<TermsConfiguration> {
...
-                AuthFactory.binder(
-                        new BasicAuthFactory<AuthenticatedUser>(
-                                new BasicAuthenticator(configuration.getCredentialsList()),
-                                'TermsApplication',
-                                AuthenticatedUser.class)))
+        environment.jersey().register(new AuthDynamicFeature(
+                new BasicCredentialAuthFilter.Builder<AuthenticatedUser>()
+                        .setAuthenticator(
+                            new BasicAuthenticator(configuration.getCredentialsList()))
+                        .setRealm('LocationApplication')
```
Comment (jaredkosanovic) [inline src/main/groovy/edu/oregonstate/mist/termsapi/TermsApplication.groovy]: LocationApplication?

Comment (gmtcrary) [reply src/main/groovy/edu/oregonstate/mist/termsapi/TermsApplication.groovy]: Oh good catch. Fixed.

File: src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package edu.oregonstate.mist.webapiskeleton
+
+import edu.oregonstate.mist.api.Application
+import edu.oregonstate.mist.api.Configuration
+import io.dropwizard.setup.Environment
+
+/**
+ * Main application class.
+ */
+class SkeletonApplication extends Application<Configuration> {
```
Comment (floppyears) [inline src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy]: Is this class needed? This should just be part of the skeleton, but shouldn't be in your api.

Comment (gmtcrary) [reply src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy]: Fixed.

**Pull request merged**